src/mds/MDLog.h

```diff
@@ -285,8 +286,8 @@ class MDLog {
 
   int64_t mdsmap_up_features = 0;
   std::map<uint64_t,std::list<PendingEvent> > pending_events; // log segment -> event list
-  ceph::mutex submit_mutex = ceph::make_mutex("MDLog::submit_mutex");
-  ceph::condition_variable submit_cond;
+  ceph::fair_mutex submit_mutex{"MDLog::submit_mutex"};
+  std::condition_variable_any submit_cond;
```

Comment:
ceph namespace in ceph_mutex.h contains mutex and condition_variable from std, can't we include condition_variable_any in ceph namespace?

Reply:
I think we should also proper handle the `seastar` case, which seems a little complicated. Since this PR just split from #41180, which has been well test before. And we also need to fix many other places after changing that, so let's do it in one separate PR later. Thanks!

Reply:
Alright, thanks!